### PR TITLE
G-API: Add new render primitives

### PR DESCRIPTION
**Pull request by TolyaTalamanov**

### CODE REVIEW

**Comment by TolyaTalamanov** [comment]
@dmatveev @smirnov-alexey

**Comment by mshabunin** [comment]
Added `cv::gapi::wip` namespace to ABI checker ignore list.

**Comment by TolyaTalamanov** [comment]
@smirnov-alexey  @AsyaPronina @dmatveev

File: modules/gapi/src/api/render_ocv.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{
+namespace gapi
+{
+namespace wip
+{
+namespace draw
+{
+
+// FIXME only for tests
+void GAPI_EXPORTS mosaic(cv::Mat mat, const cv::Rect &rect, int cellSz);
```
Comment (smirnov-alexey) [inline modules/gapi/src/api/render_ocv.hpp]: I think we should hide those due to internal usage only. Please, remove them from the header and make static/put in namespace in src. Tests are not required as you test drawPrimitivesOCV with each primitive.

**Comment by TolyaTalamanov** [comment]
Review it please @dbudniko @AsyaPronina @rgarnov @anton-potapov @dmatveev

**Comment by TolyaTalamanov** [comment]
@dmatveev

**Comment by TolyaTalamanov** [comment]
@dmatveev

File: modules/gapi/include/opencv2/gapi/render.hpp [hunk 2]
```diff
@@ -78,35 +83,96 @@ struct Line
...
+
+/**
+ * A structure to represent parameters for drawing an image
+ */
+struct Image
+{
+    int x;          //!< x-coord of image
+    int y;          //!< y-coord of image
+    cv::Mat img;    //!< Image to draw
+    cv::Mat alpha;  //!< Alpha channel for image to draw (same size and number of channels)
```
Comment (dmatveev) [inline modules/gapi/include/opencv2/gapi/render.hpp]: Alpha channel is usually a grayscale image (mask), why it has to be of the same number of channels here?

File: modules/gapi/src/api/render_ocv.cpp [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+{
+
+GAPI_OCV_KERNEL(GOCVRenderNV12, cv::gapi::wip::draw::GRenderNV12)
+{
+    static void run(const cv::Mat& y, const cv::Mat& uv, const cv::gapi::wip::draw::Prims& prims,
+                    cv::Mat& out_y, cv::Mat& out_uv)
+    {
+        /* FIXME How to render correctly on NV12 format ?
+         *
+         * Rendering on NV11 via OpenCV looks like this:
```
Comment (dmatveev) [inline modules/gapi/src/api/render_ocv.cpp]: NV11 -> NV12

Comment (TolyaTalamanov) [reply modules/gapi/src/api/render_ocv.cpp]: Fixed

File: modules/gapi/include/opencv2/gapi/render.hpp (hunk 2 above)
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/render.hpp]: @TolyaTalamanov any point on this>

Comment (smirnov-alexey) [reply modules/gapi/include/opencv2/gapi/render.hpp]: @TolyaTalamanov, could you, please, fix it? Let's discuss locally what was the problem here.

**Comment by TolyaTalamanov** [comment]
@alalek Could you merge it, please ?

**Comment by alalek** [comment]
> Merge pull request XX from TolyaTalamanov/at/new-approach-nv12-rendering

Please remove such commits. These commits create invalid references via `#` numbers.
Provide clean patches as much as possible - fixups/refactiong/etc is not really useful for PR reviewing. No merges at all, only clean rebased commits. Single squashed commit is an ideal case.

**Comment by TolyaTalamanov** [comment]
@alalek

**Comment by TolyaTalamanov** [comment]
Thanks!

**Pull request merged**